http_types/utils.py

```diff
@@ -205,15 +205,25 @@ def from_dict(obj: Any) -> RequestResponsePair:
         res_obj = obj['res']
         res = ResponseBuilder.from_dict(res_obj)
 
-        reqres = RequestResponsePair(req=req, res=res)
-        RequestResponseBuilder.validate(reqres)
+        reqres = HttpExchange(req=req, res=res)
+        HttpExchangeBuilder.validate(reqres)
         return reqres
 
     @staticmethod
-    def validate(reqres: RequestResponsePair) -> None:
+    def validate(reqres: HttpExchange) -> None:
         """Run-time typechecking for request-response pair.
 
         Arguments:
-            reqres {RequestResponsePair} -- Possible request-response object.
+            reqres {HttpExchange} -- Possible request-response object.
         """
-        check_type("request-response", reqres, RequestResponsePair)
+        check_type("request-response", reqres, HttpExchange)
+
+    @staticmethod
+    def from_jsonl(input_file: IO) -> Generator[HttpExchange, None, None]:
+        """Read Http exchanges line by line from a file-like object.
+
+        Arguments:
+            input_file: {IO} -- The input to read from.
+        """
+        for line in input_file:
+            yield HttpExchangeBuilder.from_dict(json.loads(line))
```

Comment:
Is this the same as `return (HttpExchangeBuilder.from_dict(json.loads(line)) for line in input_file)`?  Your version is more readable and therefore more pythonic, just wondering.

Reply:
It seems to be the same from what I can tell with experimenting - both creates a type that is an `isinstance(ret, types.GeneratorType)` and generates the same values.